node/src/components/consensus/era_supervisor.rs

```diff
@@ -232,12 +232,15 @@ where
             panic!("{} already exists", era_id);
         }
         self.current_era = era_id;
+        let instance_id = instance_id(&self.chainspec, state_root_hash, start_height);
 
+        validator_stakes.sort();
```

Comment:
Is it only for the logging? How will it affect the downstream of this?

Reply:
They should already be sorted at this point. (And it looks like they are.) Just wanted to make sure.

Reply:
And downstream they get sorted anyway: https://github.com/CasperLabs/casper-node/blob/b60373ae75fb13bb68f9742f31b7b48967429490/node/src/components/consensus/highway_core/validators.rs#L102

Reply:
Thanks. Are `sort()` and `sort_by(…)` guaranteed to yield the same results? I don't think so. Maybe we should sort the same way if we use it for debugging later. We need to make sure that `ValidatorIndex` matches the logged index.

Reply:
In `validators.rs`, `val0.id` is the public key. And `sort` will sort by the `(PublicKey, Motes)` tuple, so it will yield the same result (unless the same public key occurs twice, which would be a bug anyway).

Reply:
But I guess it would make sense to use `sort_by_key` in both cases, just to make sure. Will do…

Reply:
Sure, but it is equal "by coincidence". I was wondering whether we can do something more secure.